### ISSUE TITLE
Maximum call stack size exceeded when referencing external json with recursion

### ISSUE THREAD

**Issue by jose-bernard-receeve** [bcherny/json-schema-to-typescript#376]
Person.json
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "person": {
      "type": "object",
      "properties": {
        "name": { "type": "string" },
        "children": {
          "type": "array",
          "items": { "$ref": "#/definitions/person" },
          "default": []
        }
      }
    }
  },
  "type": "object",
  "properties": {
    "person": { "$ref": "#/definitions/person" }
  }
}
```

House.json
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  "properties": {
    "owner": { "$ref": "Person.json" }
  }
}
```

output
```bash
Could not generate type definitions from json-schema RangeError: Maximum call stack size exceeded
    at Object.log (/Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/utils.js:1:1)
    at generateRawType (/Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:153:13)
    at generateType (/Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:146:16)
    at /Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:273:47
    at Array.map (<anonymous>)
    at generateInterface (/Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:271:14)
    at generateRawType (/Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:168:20)
    at generateType (/Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:146:16)
    at /Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:162:28
    at generateRawType (/Users/jose/dev/Decision-Tree/node_modules/json-schema-to-typescript/dist/src/generator.js:164:15)
 ```

When I don't reference `Person.json` it works fine generating the recursive reference

Also tried this other Person schema

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": false,
  "type": "object",
    "properties": {
      "name": { "type": "string" },
      "children": {
        "type": "array",
        "items": { "$ref": "#" }
      }
  }
}
```

**Comment by htbmw** [comment]
Is there anybody who could assist with this bug.
I encounter the same and need to generate typescript from a recursive JSON schema.
@monzilnepali Could you perhaps show us on a fork where the fixes should be made?

**Comment by jan-san** [comment]
As @bcherny mentioned in #453, this issue only appears for recursive JSON schemas where the recursed schema does not have a `title`. So adding a `title` property to the schema should prevent this issue.

**Comment by htbmw** [comment]
@jan-san , thanks for the response. I checked and they all have titles.
I suspect there is some edge case scenario in my schema that causes the issue.
I will have to replicate it with an easy to share example here.

**Comment by bcherny** [comment]
Published as part of v11.0.0.
